go/vt/vttablet/tabletmanager/tm_init.go

```diff
@@ -582,6 +582,12 @@ func (tm *TabletManager) findMysqlPort(retryInterval time.Duration) {
 		if err != nil {
 			continue
 		}
+		// We need to get the action lock to make sure no one
```

Comment:
Why is the pubMu not enough?

Reply:
I guess pubMu is only for preventing data races when updating the local tablet, but this is a higher-level semantic race?

Reply:
Because `Restore` caches the tablet at the beginning and reuses it at the end. In my initial fix, I changed `Restore` to re-read the tablet before publishing it. But I remembered that this pattern exists everywhere else, which essentially requires the action lock to be the mutex for the tablet.

I think the more correct fix will be to change all call sites to not cache the tablet, but instead directly change it. This is the more subtle fix I had in mind.